File: Practice/Leetcode20_Pbm.java
import java.util.*;
public class Leetcode20_Pbm
{
    public static void main(String args[])
    {
        String s="{[(]]}";
        boolean result=isValidString(s);
        if(result)
            System.out.println("It is a valid string");
        else
            System.out.println("It is not a valid string");
    }
    public static boolean isValidString(String s)
    {
        Stack <Character> st=new Stack<Character>();
        for (int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
            if(c=='{'||c=='('||c=='[')
                st.push(c);
            else
            {
                if(st.empty())
                    return false;
                else
                {
                    char ch=st.peek();
                    if((ch=='{'&&c=='}')||(ch=='['&&c==']')||(ch=='('&&c==')')){
                        st.pop();
                    }
                    else
                        return false;
                }
            }
        }
        return st.empty();
    }
}

File: Practice/Leetcode20_Pbm.py
def isValidString(s:str)->bool:
    st=[]

    for c in s:
        if c in "{[(":
            st.append(c)
        else:
            if not st:
                return False
            else:
                peek=st[-1]
                if (c=='}' and peek=='{') or \
                   (c==']' and peek=='[') or \
                   (c==')' and peek=='('):
                   st.pop()
                else:
                    return False
    return not st

        



s="{([])}"
result=isValidString(s)
if result:
    print("It is a valid String")
else:
    print("It is not a valid string")